modules/ROOT/pages/migration-from-4x.adoc

```diff
@@ -0,0 +1,149 @@
+= Migrating from {productname} 4 to {productname} {release-version}
+:navtitle: Migrating from TinyMCE 4
+:description: Guidance for migrating from TinyMCE 4 to TinyMCE {release-version}
+:keywords: migration, considerations, premigration, pre-migration
+:release-version: 7.0
+
+
+== Overview
+
+{productname} has evolved significantly from version 4 to version {release-version}, introducing major architectural changes, modern UI improvements, enhanced performance, and better security. This guide outlines the critical breaking changes, recommended migration action steps, and top-level configuration adjustments required to upgrade from {productname} v4 to v{release-version}.
+
+== Key Changes
+
+=== UI Themes and Skins
+
+* **Removed**: modern, lightgray, and mobile themes/skins.
+* **New**: silver theme with oxide skin (supports light and dark variants). see xref:customize-ui.adoc#themes[customize-ui].
+* **Impact**: Custom v4 skins/themes are incompatible with v{release-version} and must be rewritten using the oxide skin structure.
+
+.Example:
+[source,js]
+----
+tinymce.init({
+selector: "textarea",
+skin: "oxide-dark",
+content_css: "dark",
+});
+----
+
+=== Plugin Ecosystem
+
+// The contextmenu plugin was absorbed into core, the options still works, see https://www.tiny.cloud/docs/tinymce/latest/contextmenu/. not sure why this is mentioned as removed.
+* **Removed**: `colorpicker`, `contextmenu`, `imagetools`, `legacyoutput`, `bbcode`, `fullpage`, `template`, `print`, `spellchecker`, and `tabfocus` plugins.
+* **Core Integrations**: Many plugins are now part of the core (e.g., `paste`, `hr`, `noneditable`, `table`).
+* **Premium Only**: Plugins like `spellchecker`, `tableofcontents`, `mediaembed` are now premium.
+
+Plugin Migration Examples:
+
+* `contextmenu` (Removed in v5):
+** Replace with browser-native context menus or custom menu logic.
+** Consider using link:https://www.tiny.cloud/docs/tinymce/latest/apis/tinymce.editor.ui.registry/#addContextMenu[editor.ui.registry.addContextMenu] for custom right-click actions.
+* `bbcode` (Removed in v6):
+** Replace with custom parsing or server-side processing if needed.
+* `fullpage` (Removed in v6):
+** No direct replacement. Use custom templates or server-side processing for full HTML documents.
+* `template` (Removed in v{release-version}):
+** Use premium xref:advanced-templates.adoc[advtemplate] or implement custom modal dialogs for templating.
+* `textcolor` (Removed in v6):
+** Use `forecolor` and `backcolor` buttons instead.
+
```

Comment:
Add something about toolbar buttons and menu options changed name, and maybe list the ones that changed. Does not have to be a long list, just be something simple like
> if you used the following toolbar buttons, they have now changed names. Please refer to each major version release notes for exact details: `button1`, `button2`, `somebutton`...